AppCenterData/AppCenterDataTests/MSDataTests.m

```diff
@@ -622,10 +628,257 @@ - (void)testListWhenOffline {
+  OCMStub([self.tokenExchangeMock performDbTokenAsyncOperationWithHttpClient:OCMOCK_ANY
+                                                            tokenExchangeUrl:OCMOCK_ANY
+                                                                   appSecret:OCMOCK_ANY
+                                                                   partition:kMSPartitionTest
+                                                         includeExpiredToken:YES
+                                                                reachability:OCMOCK_ANY
+                                                           completionHandler:OCMOCK_ANY])
+      .andDo(^(NSInvocation *invocation) {
+        MSGetTokenAsyncCompletionHandler getTokenCallback;
+        [invocation getArgument:&getTokenCallback atIndex:8];
+        getTokenCallback(testTokensResponse, nil);
+      });
+
+  // Mock local storage.
+  id<MSDocumentStore> localStorageMock = OCMProtocolMock(@protocol(MSDocumentStore));
+  self.sut.dataOperationProxy.documentStore = localStorageMock;
+  MSDocumentWrapper *expectedDocument = [MSDocumentWrapper new];
+  expectedDocument.pendingOperation = kMSPendingOperationDelete;
+  NSMutableArray<MSDocumentWrapper *> *localListItems = [NSMutableArray new];
+  [localListItems addObject:expectedDocument];
+  MSPage *page = [[MSPage alloc] initWithItems:localListItems];
+  MSPaginatedDocuments *expectedDocumentList = [[MSPaginatedDocuments alloc] initWithPage:page
+                                                                                partition:kMSPartitionTest
+                                                                             documentType:[MSDictionaryDocument class]
+                                                                         deviceTimeToLive:kMSDataTimeToLiveDefault
+                                                                        continuationToken:nil];
+  OCMStub([localStorageMock hasPendingOperationsForPartition:kMSPartitionTest]).andReturn(true);
+  OCMStub([localStorageMock listWithToken:testToken partition:OCMOCK_ANY documentType:OCMOCK_ANY baseOptions:OCMOCK_ANY])
+      .andReturn(expectedDocumentList);
+
+  // When
+  [MSData listDocumentsWithType:[MSDictionaryDocument class]
+                      partition:kMSPartitionTest
+              completionHandler:^(MSPaginatedDocuments *_Nonnull documents) {
+                // Then
+                XCTAssertEqualObjects(expectedDocumentList, documents);
+                [expectation fulfill];
+              }];
+
+  // Then
+  [self waitForExpectationsWithTimeout:3
+                               handler:^(NSError *_Nullable error) {
+                                 if (error) {
+                                   XCTFail(@"Expectation Failed with error: %@", error);
+                                 }
+                               }];
+}
+
+- (void)testListFromRemoteIfNotExpiredAndOnlineWithNoPendingOperation {
```

Comment:
Can we add a test to check if: we do a remote operation when we have only expired documents in the cache and we are online. I think currently this would return an empty list but it should make a remote call. Kind of a special case.

Reply:
Good catch. It wad a bug. !! Fixed it